eva-accession-pipeline/src/main/java/uk/ac/ebi/eva/accession/pipeline/steps/tasklets/reportCheck/ReportCheckTasklet.java

```diff
@@ -0,0 +1,235 @@
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.accession.pipeline.steps.tasklets.reportCheck;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.batch.core.ExitStatus;
+import org.springframework.batch.core.StepContribution;
+import org.springframework.batch.core.scope.context.ChunkContext;
+import org.springframework.batch.core.step.skip.SkipPolicy;
+import org.springframework.batch.core.step.tasklet.Tasklet;
+import org.springframework.batch.item.ExecutionContext;
+import org.springframework.batch.repeat.RepeatStatus;
+
+import uk.ac.ebi.eva.accession.pipeline.policies.InvalidVariantSkipPolicy;
+import uk.ac.ebi.eva.commons.batch.io.VcfReader;
+import uk.ac.ebi.eva.commons.core.models.factories.exception.NonVariantException;
+import uk.ac.ebi.eva.commons.core.models.pipeline.Variant;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Compares the the original VCF and accession report VCF, and logs the differences.
+ *
+ * Terminology: The original VCF contains variants, and it will be loaded into the 'variantBuffer'. The accession
+ * report VCF contains the accessioned variants, and it will be loaded into the 'accessionBuffer'.
+ *
+ * The high-level idea of the process is filling those 2 buffers, and when the same coordinates appears in both the
+ * variantBuffer and the accessionBuffer, it means that the variant in those coordinates was correctly accessioned,
+ * so it can be removed from both buffers. At the end, only variants without accessions and accessions without
+ * variants should be left in the buffers after both files were completely read.
+ *
+ * The reason why the buffers are needed is that the accession report can be unordered. In order to avoid
+ * having the original VCF completely loaded in the variantBuffer, a buffer size can be configured. However, to grant
+ * correct behaviour in worst-ordering scenario, the accessionBuffer (for yet-unmatched accessions from the accession
+ * report) has to be able to grow indefinitely.
+ *
+ * To perform self-checks, this tasklet provides the maximum size that the accessionBuffer during the execution, and
+ * also provides the number of iterations needed.
+ *
+ */
+public class ReportCheckTasklet implements Tasklet {
+
+    private static final Logger logger = LoggerFactory.getLogger(ReportCheckTasklet.class);
+
+    private static final int DEFAULT_MAX_BUFFER_SIZE = 100000;
```

Comment:
Use `chunk_size * 2`? It would make sense to keep it relatively close to the number of records read at once from the input VCF file.

Reply:
making the buffer size a function of the chunk size is a good idea. if the inter-chunk order is maintained (which I guess is what spring batch wihout concurrency does), chunk_size would be enough. But the performance of the algorithm quickly degrades if the buffer is not enough due to unorderings. Even though in theory we shouldn't get those inter-chunk unorderings, `chunk_size * 2` feels a bit tight, but as I can't provide real arguments, we can use that.

Reply:
note that this is the default, and we'll use a setter when we inject the tasklet bean